apps/api/src/report/report.util.service.ts

```diff
@@ -215,6 +215,120 @@ export class ReportUtilService {
       FROM report WHERE registration_date::date >= '${from}';
     `;
   }
+  /**
+   *
+   * @param mappedStatusesString
+   * @param from start
+   * @param to end
+   * @param BCCNM_NEW_PROCESS
+   * @returns Query string
+   */
+  reportFour(
+    mappedStatusesString: string,
+    from: string,
+    to: string,
+    BCCNM_NEW_PROCESS: boolean,
+  ): string {
+    return `
+    SELECT 
+      "status_audit"."status_id", count(status_audit.status_id)
+    FROM 
+      "ien_applicant_status_audit" "status_audit" 
+    LEFT JOIN 
+      ien_applicants applicant ON applicant.id = status_audit.applicant_id 
+    WHERE 
+      applicant.new_bccnm_process IS NOT NULL AND applicant.new_bccnm_process = ${
+        BCCNM_NEW_PROCESS ? 'TRUE' : 'FALSE'
+      } 
+      AND 
+        "status_audit"."status_id" IN 
+        (${mappedStatusesString}) 
+      AND start_date IS NOT NULL 
+      AND start_date::date >= '${from}' 
+      AND start_date::date <= '${to}' AND 
+      NOT EXISTS (
+        SELECT *
+        FROM "ien_applicant_status_audit" AS T2
+        WHERE 
+          T2.applicant_id = status_audit.applicant_id
+          AND 
+            (
+              T2.start_date > status_audit.start_date 
```

Comment:
This condition means that the milestone should be the latest one. We set an applicant's status to the latest one but don't use it. If we use applicant.status as the latest milestone, couldn't we spare this sub-query? I think we can ignore the cases that 'accept job offer' and 'withdrew from program' are not the latest.